### ISSUE TITLE
How to train the model?

### ISSUE THREAD

**Issue by XiaominLi1997** [Picsart-AI-Research/PAIR-Diffusion#8]
Hi, thanks for your nice work, if it is possible you will release the training code?

**Comment by CharlesGong12** [comment]
+1

**Comment by 761qgmpgz943** [comment]
Hi, thanks for your nice work, if it is possible you will release the training code?
It's okay not to organize the training code, I have some questions about the training process.

**Comment by vidit98** [comment]
Hi all, It will be difficult to release the training code. However, I am willing to help in specific doubts related to training. @YunChangLi yes the training code closely follows the controlnet repo. The major difference would be in dataloader where you also need to read the segmentation masks and in conditioning process. The conditioning code can be copied from the inference code already released.
